### DISCUSSION TITLE
Can I use Kingfisher to download gifs from Tenor.com API?

### DISCUSSION THREAD

**Discussion by GottaLearnThis** [onevcat/Kingfisher#1757] [Q&A]
I'm setting up an API call to Tenor.com, docs here https://tenor.com/gifapi/documentation#responseobjects-gif .

I'm setting up a struct for my returning JSON, and I'm stuck on the media part.

For "media" it says to use "[ { GIF_FORMAT : MEDIA_OBJECT } ]".

So I've got ` let media: [ Dictionary<GIF,>]   // An array of dictionaries with GIF_FORMAT as the key and MEDIA_OBJECT as the value`

Using Kingfisher, is there a way to declare and handle media objects in returning JSON?

**Comment by onevcat** [comment]
According to [the documentation](https://tenor.com/gifapi/documentation#responseobjects-media) of `MEDIA_OBJECT`, it seems that there is a `url` field containing a URL to the actual image data. Kingfisher is good at handling the image data directly (instead of processing some JSON and/or extract data from it), so I suggest you can just pass the URL to Kingfisher, letting it download from the URL and show it in an image view.

**Reply by GottaLearnThis** [reply]
Thanks I may try that
